api/ttn/lorawan/v3/gateway_services.proto

```diff
@@ -163,3 +164,19 @@ message PullGatewayConfigurationRequest {
 service GatewayConfigurator {
   rpc PullConfiguration(PullGatewayConfigurationRequest) returns (stream Gateway);
 }
+
+message AssertGatewayRightsRequest {
+  repeated GatewayIdentifiers gateway_ids = 1 [
+    (validate.rules).repeated.min_items = 1,
+    (validate.rules).repeated.max_items = 20
+  ];
+  Rights required = 2 [(validate.rules).message.required = true];
+}
+
+service GatewayBatchAccess {
```

Comment:
Let's mark this RPC as experimental as we fiddle more with these operations for now. We can graduate them later on.